src/DynamoCoreWpf/Views/Core/DynamoView.xaml.cs

```diff
@@ -963,7 +963,16 @@ void DynamoView_KeyDown(object sender, KeyEventArgs e)
 
         void DynamoView_KeyUp(object sender, KeyEventArgs e)
         {
-            if (e.Key == Key.Escape)
+            if (e.Key == Key.Escape && dynamoViewModel.WatchEscapeIsDown)
+            {
+                dynamoViewModel.WatchEscapeIsDown = false;
+                dynamoViewModel.EscapeCommand.Execute(null);
+            }
```

Comment:
Is the addition `dynamoViewModel.WatchEscapeIsDown` an optimization so that we don't unnecessarily execute the body of this `if` condition? Or is it mean to address some other scenarios that we do not know about?